From 667db73d465dd0eaabdae312489590901c7ca3af Mon Sep 17 00:00:00 2001
From: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
Date: Wed, 27 Oct 2021 19:54:38 -0500
Subject: [PATCH 1/1] drivers/mtd/spi-nor: Add support for Gigadevice flash
 chip
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit

Adds support for Gigadevice flash chip with JEDEC ID "c8 47 1a".

Signed-off-by: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
---
 drivers/mtd/spi-nor/gigadevice.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/spi-nor/gigadevice.c b/drivers/mtd/spi-nor/gigadevice.c
index 447d84bb2128..d7b128b6f573 100644
--- a/drivers/mtd/spi-nor/gigadevice.c
+++ b/drivers/mtd/spi-nor/gigadevice.c
@@ -49,7 +49,9 @@ static const struct flash_info gigadevice_parts[] = {
 			   SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
 			   SPI_NOR_4B_OPCODES | SPI_NOR_HAS_LOCK |
 			   SPI_NOR_HAS_TB | SPI_NOR_TB_SR_BIT6)
-		.fixups = &gd25q256_fixups },
+			.fixups = &gd25q256_fixups },
+	{ "gd25b512", INFO(0xc8471a, 0, 64 * 1024, 512,
+			   SECT_4K | SPI_NOR_QUAD_READ) },
 };

 const struct spi_nor_manufacturer spi_nor_gigadevice = {
--
2.17.1

